Explorer/Assets/DCL/SDKEntityTriggerArea/Components/SDKEntityTriggerAreaComponent.cs

```diff
@@ -0,0 +1,154 @@
+using CrdtEcsBridge.Physics;
+using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using ECS.Unity.Transforms.Components;
+using SceneRunner.Scene;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.SDKEntityTriggerArea.Components
+{
+    public enum SDKEntityTriggerAreaMeshType
+    {
+        BOX,
+        SPHERE
+    }
+
+    public struct SDKEntityTriggerAreaComponent : IDirtyMarker
+    {
+        private static readonly IReadOnlyCollection<Collider> EMPTY_COLLECTION = Array.Empty<Collider>();
+        public SDKEntityTriggerArea? monoBehaviour { get; private set; }
+
+        private readonly bool targetOnlyMainPlayer;
+        private bool hasMonoBehaviour;
+
+        public Vector3 AreaSize { get; private set; }
+        public SDKEntityTriggerAreaMeshType MeshType { get; private set; }
+        public ColliderLayer LayerMask { get; private set; }
+        public bool IsDirty { get; set; }
+
+        public readonly IReadOnlyCollection<Collider> EnteredEntitiesToBeProcessed => hasMonoBehaviour
+            ? monoBehaviour!.EnteredEntitiesToBeProcessed
+            : EMPTY_COLLECTION;
+
+        public readonly IReadOnlyCollection<Collider> ExitedEntitiesToBeProcessed => hasMonoBehaviour
+            ? monoBehaviour!.ExitedEntitiesToBeProcessed
+            : EMPTY_COLLECTION;
+
+        public readonly IReadOnlyCollection<Collider> CurrentEntitiesInside => hasMonoBehaviour
+            ? monoBehaviour!.CurrentEntitiesInside
+            : EMPTY_COLLECTION;
+
+        public SDKEntityTriggerAreaComponent(
+            Vector3 areaSize,
+            bool targetOnlyMainPlayer = false,
+            SDKEntityTriggerArea? monoBehaviour = null,
+            SDKEntityTriggerAreaMeshType meshType = SDKEntityTriggerAreaMeshType.BOX,
+            ColliderLayer layerMask = ColliderLayer.ClPlayer)
+        {
+            AreaSize = areaSize;
+            this.MeshType = meshType;
+            this.LayerMask = layerMask;
+            this.targetOnlyMainPlayer = targetOnlyMainPlayer;
+
+            this.monoBehaviour = monoBehaviour;
+            hasMonoBehaviour = monoBehaviour != null;
+
+            IsDirty = true;
+        }
+
+        public void TryAssignArea(IComponentPool<SDKEntityTriggerArea> pool, Transform mainPlayerTransform, TransformComponent transformComponent)
+        {
+            if (hasMonoBehaviour)
```

Comment:
Should this logic be after checking `IsDirty` or there is a special hidden case?

Reply:
You are right that is strange, my old original implementaiton didn't have that, it was introduced at https://github.com/decentraland/unity-explorer/commit/d1b0c1505184c96ee049c65cf7e18c73a6b4c2f4, I'll see if it makes sense to have that code before the `IsDirty` check.

Reply:
I found the root cause in [that old PR](https://github.com/decentraland/unity-explorer/pull/2859), that code was being run before the `IsDirty` to handle re-activating the colliders when the user re-enter the scene... but it was being handled in a very obscure and confusing way, so I refactored and I think it is much better now, WDYT? handled at https://github.com/decentraland/unity-explorer/pull/5244/commits/c38d84d8b3b8c6f6595e12cae374c96b4d0ed227